pkg/controller/core/workload_controller.go

```diff
@@ -152,10 +152,37 @@ func (r *WorkloadReconciler) Reconcile(ctx context.Context, req ctrl.Request) (c
 	ctx = ctrl.LoggerInto(ctx, log)
 	log.V(2).Info("Reconciling Workload")
 
-	// If a deactivated workload is re-activated, we need to reset the RequeueState.
-	if wl.Status.RequeueState != nil && ptr.Deref(wl.Spec.Active, true) && workload.IsEvictedByDeactivation(&wl) {
-		wl.Status.RequeueState = nil
-		return ctrl.Result{}, workload.ApplyAdmissionStatus(ctx, r.client, &wl, true)
+	if ptr.Deref(wl.Spec.Active, true) {
```

Comment:
I'm thinking we should move this block below L188 (handling of deletion), and maybe even L192 (handling of workload finished). The reason being, that now we handle here PodsReadyTimeout, and this may result in waiting potentially arbitrarily long for requeue, the waiting times grow exponentially like 10s, 20s, 40s... . 

However, it is possible a user deletes the workload in the meanwhile, and we are not able to remove finalizer, and we unnecessarily wait for the requeue. Similarly, a user may mark the workload as finished. 

Could you check if this is the case, then please fix (not highest priority, so could be follow up), plus add a test?

Reply:
Yes, you right. Moved it and added tests.